### PR TITLE
[Bugfix] Drill-down (cascading) forms in Lizmap based on QGIS expression

### PR DESCRIPTION
**Pull request by rldhont**
The criteriaFrom for the drill-down based on QGIS Expression was not performed.

Logs:
```
Invalid argument supplied for foreach()
  /srv/lizmap/lizmap/modules/lizmap/classes/qgisFormValueRelationDynamicDatasource.class.php	40
```

* Funded by 3liz

### CODE REVIEW

**Comment by mdouchin** [comment]
Looks good to me. Do you think a simple test could be done with 2 XML: one with a criteriaFrom, another one without ?

**Review by mdouchin — CHANGES_REQUESTED** [review]
I tested

* *Check the default form* -> OK
* *Check the drill-down fields*
    - Filter wrong when changing the field *Parent field* To correct it, I replaced `getCriteriaFromExpression` by `getCurrentValueCriteriaFromExpression` in the class `qgisValueRelationDynamicDatasource.class.php` **line 40**
* *Check the field available values filtered by geometry* -> it does not work neither. @rldhont knows what to change to make it pass

**Review by mdouchin — APPROVED** [review]

File: tests/qgis-projects/tests/form_edition_value_relation_field.qgs.cfg [hunk 1]
```diff
@@ -0,0 +1,237 @@
+{
+    "metadata": {
+        "lizmap_plugin_version": 30209,
```
Comment (nboisteault) [inline tests/qgis-projects/tests/form_edition_value_relation_field.qgs.cfg:3]: @rldhont Your lizmap plugin version is outdated.

File: tests/qgis-projects/tests/form_edition_value_relation_field.qgs.cfg [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+        "lizmap_web_client_target_version": "3.3"
```
Comment (nboisteault) [inline tests/qgis-projects/tests/form_edition_value_relation_field.qgs.cfg:4]: It should be 3.4

**Review by nboisteault — CHANGES_REQUESTED** [review]

File: tests/qgis-projects/tests/form_edition_value_relation_field.qgs.cfg (hunk 2 above)
Comment (rldhont) [reply tests/qgis-projects/tests/form_edition_value_relation_field.qgs.cfg]: updated

File: tests/qgis-projects/tests/form_edition_value_relation_field.qgs.cfg (hunk 1 above)
Comment (rldhont) [reply tests/qgis-projects/tests/form_edition_value_relation_field.qgs.cfg]: updated

**Pull request merged**